### PR TITLE
Update ring_buffer.h

### PR DESCRIPTION
**Pull request by hardlyli**
# Thank you for your contribution

You are welcome to open PR, but they are used for discussion only. All
patches must eventually go to the openvpn-devel mailing list for review:

* https://lists.sourceforge.net/lists/listinfo/openvpn-devel

Please send your patch using [git-send-email](https://git-scm.com/docs/git-send-email). For example to send your latest commit to the list:

    $ git send-email --to=openvpn-devel@lists.sourceforge.net HEAD~1

For details, see these Wiki articles:

* https://community.openvpn.net/openvpn/wiki/DeveloperDocumentation
* https://community.openvpn.net/openvpn/wiki/Contributing

### CODE REVIEW

File: src/openvpn/ring_buffer.h [hunk 1]
```diff
@@ -106,7 +106,9 @@ register_ring_buffers(HANDLE device,
     DWORD bytes_returned;
 
     ZeroMemory(&rr, sizeof(rr));
-
+/*
+Fills a block of memory with zeros.
```
Comment (schwabe) [inline src/openvpn/ring_buffer.h:110]: That comment doesn't add any value.

File: src/openvpn/ring_buffer.h [hunk 2]
```diff
@@ -117,7 +119,19 @@ register_ring_buffers(HANDLE device,
...
+    BOOL DeviceIoControl(
+  [in]                HANDLE       hDevice,
+  [in]                DWORD        dwIoControlCode,
+  [in, optional]      LPVOID       lpInBuffer,
+  [in]                DWORD        nInBufferSize,
+  [out, optional]     LPVOID       lpOutBuffer,
+  [in]                DWORD        nOutBufferSize,
+  [out, optional]     LPDWORD      lpBytesReturned,
+  [in, out, optional] LPOVERLAPPED lpOverlapped
+);
```
Comment (schwabe) [inline src/openvpn/ring_buffer.h:133]: I wonder why we should document library function in OpenVPN. Shouldn't a good IDE also allow you to lookup thiee?

**Comment by dsommers** [comment]
@hardlyli Can you please provide some reasoning why this pull request is important to you?  How does these changes improve the code?

File: src/openvpn/ring_buffer.h (hunk 2 above)
Comment (ordex) [reply src/openvpn/ring_buffer.h:133]: The API documentation belongs to where the function is defined/declared (i.e. the SDK), not in the user.

**Pull request closed without merging**